### PR TITLE
 Run test-make-requirements.sh after other tests

### PR DESCRIPTION
**Pull request by millerdev**
Didn't have time to verify that running `test-make-requirements.sh` in docker will actually work (not sure about the git commands), so the tests might fail in which case I'll find another way to do this.

Update: it failed. Will have to tweak this a bit.

@emord

### CODE REVIEW

File: scripts/test-make-requirements.sh [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/usr/bin/env bash
 set -e
 
-pip install pip-tools>=3.8
+# This is a dirty test. It can leave some files in a different state
+# than they were when it started.
```
Comment (emord) [inline scripts/test-make-requirements.sh]: Would running `git checkout requirements/ requirements-py3/` before `exit 1` in the else statement fix this?

**Review by emord — APPROVED** [review]

**Comment by millerdev** [comment]
It took a bunch of iterations, but should be g2g now. Wait for build on second-to-last commit.

File: scripts/pip-post-compile.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-bash scripts/_vendor/pip-post-compile.sh "$@"
```
Comment (emord) [inline scripts/pip-post-compile.sh]: should this PR also remove this script from _vendor?

**Review by emord — APPROVED** [review]

File: scripts/pip-post-compile.sh (hunk 2 above)
Comment (millerdev) [reply scripts/pip-post-compile.sh]: it's ignored

https://github.com/dimagi/commcare-hq/blob/a7a1e0beb3da733b7634de123049953a4b845584/.gitignore#L9

Comment (emord) [reply scripts/pip-post-compile.sh]: ah. that makes sense after looking at the old Makefile again. thanks for this change

**Pull request merged**